django/contrib/postgres/fields/array.py

```diff
@@ -157,6 +157,12 @@ def get_transform(self, name):
         else:
             return SliceTransformFactory(start, end)
 
+    def slice_expression(self, expression, start, length):
+        if length is None:
+            # 2147483647 being maximum 32bit integer
+            return SliceTransform(start, 2_147_483_647, expression)
+        return SliceTransform(start, min(2_147_483_647, start + length - 1), expression)
```

Comment:
I was unable to find why we're doing the 32-bit thing here... I can't find anything in the previous PRs, the ticket, or the PostgreSQL docs... Maybe I just missed it.

As this is for PostgreSQL arrays, we can just omit the `length` when it is `None`. See [here](https://dbfiddle.uk/?rdbms=postgres_13&fiddle=1549c558228b821fe23de68bfdab81d5) for an example. That is assuming we can do `SliceTransform(start, '', expression)`.

I'm not sure we need to handhold if the "end" value is too large. If PostgreSQL blows up then we should just let that raise out. It's pretty unlikely anyway.

```suggestion
        # If length is not provided, don't specify an end to slice to the end of the array.
        end = '' if length is None else start + length - 1
        return SliceTransform(start, end, expression)
```